tests/commands/test_store_commands.py

```diff
@@ -1331,12 +1331,205 @@ def test_status_with_resources_and_branches(caplog, store_mock, config):
+    assert expected == [rec.message for rec in caplog.records]
+
+
+def test_status_multiplebases_everything_combined(caplog, store_mock, config):
+    """Multiple bases with several other modifiers, just a sanity check."""
+    caplog.set_level(logging.INFO, logger="charmcraft.commands")
+
+    other_base = Base(architecture="16b", channel="1", name="xz")
+    tstamp = dateutil.parser.parse("2020-07-03T20:30:40Z")
+    resource = Resource(name="testres", optional=True, revision=1, resource_type="file")
+    channel_map = [
+        _build_release(revision=7, channel="latest/candidate"),
+        _build_release(revision=156, channel="latest/edge"),
+        _build_release(revision=7, channel="latest/candidate/br1", expires_at=tstamp),
+        _build_release(revision=7, channel="latest/stable", base=other_base),
+        _build_release(revision=80, channel="latest/beta", base=other_base),
+        _build_release(
+            revision=99,
+            channel="latest/beta/br2",
+            base=other_base,
+            expires_at=tstamp,
+            resources=[resource],
+        ),
+        _build_release(revision=7, channel="2.0/candidate"),
+        _build_release(revision=80, channel="2.0/beta"),
+        _build_release(revision=7, channel="2.0/stable", base=other_base),
+        _build_release(revision=80, channel="2.0/edge", base=other_base),
+        _build_release(
+            revision=80, channel="2.0/edge/foobar", base=other_base, expires_at=tstamp
+        ),
+    ]
+    channels = _build_channels() + _build_channels(track="2.0")
+    channels.extend(
+        [
+            Channel(
+                name="latest/candidate/br1",
+                fallback="latest/candidate",
+                track="latest",
+                risk="candidate",
+                branch="br1",
+            ),
+            Channel(
+                name="latest/beta/br2",
+                fallback="latest/beta",
+                track="latest",
+                risk="beta",
+                branch="br2",
+            ),
+            Channel(
+                name="2.0/edge/foobar",
+                fallback="2.0/edge",
+                track="2.0",
+                risk="edge",
+                branch="foobar",
+            ),
+        ]
+    )
+    revisions = [
+        _build_revision(revno=7, version="v7"),
+        _build_revision(revno=80, version="2.0"),
+        _build_revision(revno=156, version="git-0db35ea1"),
+        _build_revision(revno=99, version="weird"),
+    ]
+    store_mock.list_releases.return_value = (channel_map, channels, revisions)
+
+    args = Namespace(name="testcharm")
+    StatusCommand("group", config).run(args)
+
+    assert store_mock.mock_calls == [
+        call.list_releases("testcharm"),
+    ]
+
+    expected = [
+        "Track    Base                  Channel        Version       Revision    Resources     Expires at",  # NOQA
+        "latest   ubuntu 20.04 (amd64)  stable         -             -           -",
```

Comment:
If a charm supports multiple arches, does it get its own entry even though its the same revision?  Or can "Base" support multiple arch?

Reply:
It gets its own entry, you end up getting everything alphabetically sorted and easy to find. In the future we may want to change it, if real usage proves better to "combine" it.